src/globus_cli/commands/timer/_common.py

```diff
@@ -90,3 +107,81 @@ def isoformat_to_local(
     """,
     flags=re.VERBOSE,
 )
+
+
+def parse_timedelta(s: str) -> datetime.timedelta:
+    matches = timedelta_regex.match(s)
+    if not matches:
+        raise ValueError(f"couldn't parse timedelta from string: {s}")
+    groups = {k: int(v) for k, v in matches.groupdict(0).items()}
+    # timedelta accepts kwargs for units up through days, have to convert weeks
+    groups["days"] += groups.pop("weeks", 0) * 7
+    return datetime.timedelta(**groups)
+
+
+def read_csv(
+    file_name: str,
+    fieldnames: List[str],
+    comment_char: str = "#",
+) -> Generator[Dict[str, Union[str, bool]], None, None]:
+    def decomment(f):
+        for row in f:
+            if not row.startswith(comment_char):
+                yield row
+
+    def transform_val(k: str, v: str) -> Union[str, bool]:
+        v = v.strip()
+        if k == "recursive":
+            try:
+                return bool(strtobool(v))
```

Comment:
If we keep this (rather than following the suggestion to use the `--batch` processing below), then I'd like to see us not use `distutils` because it's deprecated. I can read this section more carefully if we go down this route.